templates/OperationRequestBodySnippet.stencil

```diff
@@ -31,11 +31,23 @@ if ({{ op.params.body.param.name }} != nil)
 }
 {% elif op.params.body.strategy == "byteArray" %}
 let requestBody = {{ op.params.body.param.name }}
-{% elif op.params.body.strategy == "unixTime" %}
+{% elif op.params.body.strategy == "date" or op.params.body.strategy == "dateTime"  or  op.params.body.strategy == "dateTimeRfc1123"  or op.params.body.strategy == "unixTime" %}
 let encoder = JSONEncoder()
+{% if op.params.body.strategy == "date" %}
+let dateFormatter = DateFormatter()
+dateFormatter.dateFormat = "yyyy-MM-dd"
+encoder.dateEncodingStrategy = .formatted(dateFormatter)
+{% elif op.params.body.strategy == "dateTime" %}
+let dateFormatter = Date.AzureISO8601DateFormatter()
+encoder.dateEncodingStrategy = .formatted(dateFormatter)
+{% elif op.params.body.strategy == "dateTimeRfc1123" %}
+let dateFormatter = Date.AzureRfc1123DateFormatter()
+encoder.dateEncodingStrategy = .formatted(dateFormatter)
+{% elif op.params.body.strategy == "unixTime" %}
 encoder.dateEncodingStrategy = .secondsSince1970
+{% endif %}
```

Comment:
I feel like his definitely would benefit from indentation since it looks like there's a nested if instead here.

Reply:
Done.